activesupport/lib/active_support/fragment_cache_key.rb

```diff
@@ -0,0 +1,9 @@
+module FragmentCacheKey
+  # Given a key (as described in ActionController::Caching::Fragments.expire_fragment),
+  # returns a key suitable for use in reading, writing, or expiring a
+  # cached fragment. All keys are prefixed with <tt>views/</tt> and uses
+  # ActiveSupport::Cache.expand_cache_key for the expansion.
+  def fragment_cache_key(key)
+    ActiveSupport::Cache.expand_cache_key(key.is_a?(Hash) ? url_for(key).split("://").last : key, :views)
```

Comment:
Sure, I'll dupe it in `CollectionCaching` instead.

Honestly, I don't know. This should go into Action View where the other caching helpers are.